docs/spec_draft.md

```diff
@@ -3731,6 +3819,50 @@ where `pred_val = rank(pred) == 0 ? pred : pred[i0, ..., iR-1]`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.send
+
+### Semantics
+
+Sends data `inputs` from one partition to a [stablehlo.recv](#stablehlorecv)
+instruction, in another partition, that shares the same channel handle pair
+{`channel_id`, `channel_type`}.
+
+If `is_host_transfer`, an optional input with default value as `fasle`, is
+`true`, then the operation transfers data to the host.
+
+### Inputs
+
+| Name               | Type                                                               |
+|--------------------|--------------------------------------------------------------------|
+| `inputs`           | variadic number of tensors of any supported type                   |
+| `token`            | `token`                                                            |
+| `channel_id`       | constant of type `si64`                                            |
+| `channel_type`     | enum of `DEVICE_TO_DEVICE`, `DEVICE_TO_HOST`, and `HOST_TO_DEVICE` |
+| `is_host_transfer` | constant of type `i1`                                              |
+
+### Outputs
+
+| Name      | Type  |
+|-----------|-------|
+| `results` | token |
+
+### Constraints
+  * (C1) size(`inputs`) $\gt$ 0 if `is_host_transfer` $=$ `true`
```

Comment:
Where is this constraint coming from? Did you perhaps mean `= 1` based on [this excerpt from xla_builder.cc](https://github.com/tensorflow/tensorflow/blob/8f75790b06acde30737a8698e4090bb52833901a/tensorflow/compiler/xla/client/xla_builder.cc#L3742-L3746)?

Overall, this seems to me like a rare case of a limitation of HLO that we don't have to emulate, especially given that there's a TODO to fix it on the HLO side.

Reply:
Actually I meant `>=2` based on the excerpt you shared.